stubs/CoreGenericFunctions.phpstub

```diff
@@ -1482,11 +1482,11 @@ function bin2hex(string $string): string {}
  *
  * @param resource|null $context
  *
- * @return ($associative is 0 ? list<string> : array<string, string|list<string>>)|false
+ * @return ($associative is false ? list<string> : array<string, string|list<string>>)|false
  *
  * @psalm-taint-sink ssrf $url
  */
-function get_headers(string $url, int $associative = 0, $context = null) : array|false {}
+function get_headers(string $url, bool $associative = false, $context = null) : array|false {}
```

Comment:
This will be harder to handle because we don't have conditional params so you can't really have different types for different versions. Could you try to leave this stub as it is here and try to duplicate it in PHP80.phpstub ?
The hope is that the second stub will override the first but I'm not sure it will.

Reply:
Seems to work with my local testing...  I test my example file in the bug report with   php ../psalm --php-version=7.4.0 --no-cache and  php ../psalm --php-version=8.0.0--no-cache with both int and bool and correct correct results for each version. 

I am happy to also add a test case for this, i am uncertain where in the tests that should live... If you can provide some guidance I am to add that.

Reply:
You can add your tests on FunctionCallTest::providerValidCodeParse :)

Reply:
Thanks for the pointer. Test are added.

Reply:
Wan you add the expected return type in the assertion array part of the test?

In PHP 7 and 8, you should get slightly different types, this should help us see if the return behaves correctly too

Reply:
Assertions are added.